src/v/coproc/pacemaker.cc

```diff
@@ -68,17 +68,20 @@ pacemaker::pacemaker(unresolved_address addr, ss::sharded<storage::api>& api)
 ss::future<> pacemaker::start() {
     co_await ss::recursive_touch_directory(offsets_snapshot_path().string());
     auto ncc = co_await recover_offsets(_offs.snap, _shared_res.api.log_mgr());
-    _ntps = std::move(ncc);
+    std::exchange(_ntps, std::move(ncc));
     if (!_offs.timer.armed()) {
         _offs.timer.arm(_offs.duration);
     }
 }
 
 ss::future<> pacemaker::reset() {
     _offs.timer.cancel();
+    ntp_context_cache ncc;
+    std::exchange(ncc, std::move(_ntps));
```

Comment:
prefer `ncc = std::exchange(_ntps, {})`?

Reply:
Ah neat didn't understand your initial comment, didn't know std::exchange returns the old val